subsys/net/lib/coap/coap_client.c

```diff
@@ -320,6 +328,7 @@ int coap_client_req(struct coap_client *client, int sock, const struct sockaddr
 	reset_internal_request(internal_req);
 
 	if (k_mutex_lock(&client->send_mutex, K_NO_WAIT)) {
+		LOG_DBG("Could not immediately lock mutex");
```

Comment:
Perhaps `send_mutex` instead of just `mutex`. It might make finding the source of this message a little easier